tests/neg/i2473.scala

```diff
@@ -0,0 +1,7 @@
+trait Foo { trait Inner }
+trait Bar { foo: Foo =>
+  type Inner <: foo.Inner // error: type Inner cannot have the same name as trait Inner in trait Foo -- class definitions cannot be overridden
```

Comment:
In fact, I don't think that shadowing is the right term either, consider:
```scala
trait Foo {
  val x: Int = 1
}

trait Bar { foo: Foo =>
  val x: String = ""
}
```
In both scalac and dotty, this code compiles, but if you try to make an instance of `Bar`, it fails:
```scala
object Test {
  def main(args: Array[String]): Unit = {
    new Foo with Bar {}
  }
}
```

```scala
-- Error: try/i2473.scala:14:4 -------------------------------------------------
14 |    new Foo with Bar {}
   |    ^
   |   class $anon inherits conflicting members:
   |     value x in trait Foo of type Int  and
   |     value x in trait Bar of type String
   |   (Note: this can be resolved by declaring an override in class $anon.)
```